packages/api/endpoints/granules.js

```diff
@@ -277,6 +301,25 @@ const handlePutAction = async (
   const body = req.body;
   const action = body.action;
 
+  if (!action) {
+    if (req.body.granuleId === req.params.granuleName) {
+      return patchGranule(req, res);
+    }
+    return res.boom.badRequest(
+      `input :granuleName (${req.params.granuleName}) must match body's granuleId (${req.body.granuleId})`
+    );
+  }
+
+  const pgGranule = await getUniqueGranuleByGranuleId(
+    knex,
+    req.params.granuleName,
+    granulePgModel
+  );
+
+  const collectionPgModel = new CollectionPgModel();
+  const pgCollection = await collectionPgModel.get(knex, {
+    cumulus_id: pgGranule.collection_cumulus_id,
+  });
   const apiGranule = await translatePostgresGranuleToApiGranule({
```

Comment:
At first glance.... it looks like this dupes a lot of logic between patchByGranuleId and patch - is there a reason to duplicate the code outside of the collection lookup/unique granule constraints code?     

* Consider DRYing this up a bit

Reply:
:100:  Thanks for doing that - I grok the scoping concern, but having that code duplicated feels risky in terms of downstream work/tests.